homeassistant/components/onvif/config_flow.py

```diff
@@ -0,0 +1,349 @@
+            if not self.device_id:
+                devicemgmt = camera.create_devicemgmt_service()
+                network_interfaces = await devicemgmt.GetNetworkInterfaces()
+                for interface in network_interfaces:
+                    if interface.Enabled:
+                        self.device_id = interface.Info.HwAddress
+
+            if self.device_id is None:
+                return self.async_abort(reason="no_mac")
+
+            await self.async_set_unique_id(self.device_id)
+            self._abort_if_unique_id_configured(
+                updates={
+                    CONF_HOST: self.onvif_config[CONF_HOST],
+                    CONF_PORT: self.onvif_config[CONF_PORT],
+                    CONF_NAME: self.onvif_config[CONF_NAME],
+                }
+            )
+
+            # Shortcut if profiles already configured
+            if self.onvif_config.get(CONF_PROFILE):
+                return await self.async_step_configure_profile()
+
+            media_service = camera.create_media_service()
+            profiles = await media_service.GetProfiles()
+            LOGGER.debug("Media Profiles %s", pformat(profiles))
+            for i, profile in enumerate(profiles):
+                if profile.VideoEncoderConfiguration.Encoding != "H264":
+                    continue
+                self.profiles[i] = {
+                    "name": profile.Name,
+                    "resolution": "%dx%d"
+                    % (
+                        profile.VideoEncoderConfiguration.Resolution.Width,
+                        profile.VideoEncoderConfiguration.Resolution.Height,
+                    ),
+                }
+
+            if not self.profiles:
+                return self.async_abort(reason="no_h264")
+
+            self.onvif_config[CONF_PROFILE] = {}
+
+            return await self.async_step_configure_profile()
+            # return await self._create_entry()
+        except exceptions.ONVIFError as err:
+            LOGGER.error(
+                "Couldn't setup ONVIF device '%s'. Error: %s",
+                self.onvif_config[CONF_NAME],
+                err,
+            )
+            return self.async_abort(reason="onvif_error")
+
+        except Fault:
+            errors["base"] = "connection_failed"
+
+        return self.async_show_form(step_id="auth", errors=errors)
+
+    async def async_step_configure_profile(self, user_input=None):
+        """Allow the user to configure ONVIF profiles."""
+        if user_input is not None:
+            self.onvif_config[CONF_PROFILE][self.active_profile] = user_input[
+                CONF_INCLUDE
+            ]
+            self.profiles.pop(self.active_profile)
+            self.active_profile = None
+
+        # find the next unconfigured profile
+        for i, profile in self.profiles.items():
+            self.active_profile = i
+            return self.async_show_form(
+                step_id="configure_profile",
```

Comment:
Why don't you use a `cv.multi_select` to allow users to pick profiles to include ? Also would it make sense to pick the one with the highest resolution by default and allow configuring others in the options flow ?

Reply:
`cv.multi_select` sounds perfect! Didn’t use it because I didn’t know about it, so tried to get fancy!

Regarding not simply defaulting a profile, see other comments about how NVRs work.